File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/trading-engine-server/TradingEngineServerHostBuilder.cpp
#include "TradingEngineServerHostBuilder.h"
#include <iostream>
#include "../logging/ConsoleLogger.h"

using namespace std;

shared_ptr<TradingEngineServer> TradingEngineServerHostBuilder::BuildTradingEngineServer() {
    // Start with configuration
    auto config = std::make_shared<TradingEngineServerConfiguration>();
    // Set configuration options here (e.g., from a file or environment variables)
    config->GetTradingEngineServerSettings().Port = 8080; // Example: setting the port

    auto logger = make_shared<ConsoleLogger>();
    // services.AddSingleton<ITextLogger, >();
    /*
    * 
    * 
    ** /

    // Create the TradingEngineServer
    auto tradingEngineServer = make_shared<TradingEngineServer>(logger, config);

    // Return the configured server
    return tradingEngineServer;
}

File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/tradingserverengine/TradingEngineServerHostBuilder.java
package com.fintech.algotrading.tradingserverengine;

import com.fintech.algotrading.logging.ConsoleLogger;

//TradingEngineServerHostBuilder
public class TradingEngineServerHostBuilder {
 public static TradingEngineServer buildTradingEngineServer() {
     // Start with configuration
     TradingEngineServerConfiguration config = new TradingEngineServerConfiguration();
     // Set configuration options here (e.g., from a file or environment variables)
     config.getTradingEngineServerSettings().setPort(8080); // Example: setting the port

     // Create a logger
     ConsoleLogger consoleLogger = new ConsoleLogger();

     // Create the TradingEngineServer
     TradingEngineServer tradingEngineServer = new TradingEngineServer(consoleLogger, config);

     // Return the configured server
     return tradingEngineServer;
 }
}